src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java

```diff
@@ -39,4 +63,94 @@ public WarehouseDto save(WarehouseDto dto) {
     public void delete(Long id) {
 
     }
+
+    @Override
+    @Transactional
+    public void acceptItems(final ApplicationDto appDto, final Long warehouseId) {
+        lock.lock();
```

Comment:
May we use synchronization on application id?

Reply:
sounds reasonable, but lets think about other case, if someone tries to update that Application at the same time, is it possible that he tries to update outdated entries?